aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1074,7 +1074,11 @@ impl AptosVM {
             bcs::to_bytes(&payload).map_err(|_| invariant_violation_error())?
         } else {
             // Default to empty bytes if payload is not provided.
-            bcs::to_bytes::<Vec<u8>>(&vec![]).map_err(|_| invariant_violation_error())?
+            if self.features().is_multisig_v2_fix_enabled() {
+                vec![]
+            } else {
+                bcs::to_bytes::<Vec<u8>>(&vec![]).map_err(|_| invariant_violation_error())?
```

Comment:
can this ever fail? makes sense with gating but curious if we can just have `vec![]` here?

Reply:
The vector should represent the serialization of the provided payload. Defining it as `vec![]` for cases where there is no payload is more convenient. Using `bcs::to_bytes::<Vec<u8>>(&vec![])` does not yield an empty vector but rather `[1]` (or something similar), which is not an intuitive representation of the absence of a payload.